bindings/wasm/src/tangle/resolver.rs

```diff
@@ -212,6 +228,57 @@ impl WasmResolver {
     // WARNING: this does not validate the return type. Check carefully.
     Ok(promise.unchecked_into::<PromiseResolvedDocument>())
   }
+
+  /// Verifies a `Presentation`.
+  ///
+  /// ### Important
+  /// See `PresentationValidator::validate` for information about which properties get
+  /// validated and what is expected of the optional arguments `holder` and `issuer`.
+  ///
+  /// ### Resolution
+  /// The DID Documents for the `holder` and `issuers` are optionally resolved if not given.
+  /// If you already have up-to-date versions of these DID Documents, you may want
+  /// to use `PresentationValidator::validate`.
+  /// See also `Resolver::resolvePresentationIssuers` and `Resolver::resolvePresentationHolder`.
+  ///
+  /// ### Errors
+  /// Errors from resolving the holder and issuer DID Documents, if not provided, will be returned immediately.
+  /// Otherwise, errors from validating the presentation and its credentials will be returned
+  /// according to the `fail_fast` parameter.
+  #[wasm_bindgen(js_name = verifyPresentation)]
+  pub fn verify_presentation(
+    &self,
+    presentation: &WasmPresentation,
+    options: &WasmPresentationValidationOptions,
+    fail_fast: WasmFailFast,
+    holder: OptionResolvedDocument,
+    issuers: OptionArrayResolvedDocument,
+  ) -> Result<PromiseVoid> {
```

Comment:
It's a bit odd to have the order of parameters differ between the Rust and Wasm versions. We could have the same ordering and change `fail_fast` to take on option to facilitate omitting parameters in Javascript?

```suggestion
  pub fn verify_presentation(
    &self,
    presentation: &WasmPresentation,
    options: &WasmPresentationValidationOptions,
    holder: OptionResolvedDocument,
    issuers: OptionArrayResolvedDocument,
    fail_fast: Option<WasmFailFast>,
  ) -> Result<PromiseVoid> {
```

Reply:
I agree that the ordering is not ideal, but I am still against making `fail_fast` optional as there is no canonical default in my opinion. It is also a bit odd to have `Some(FailFast::AllErrors)` as an argument on the Rust side. I would be fine adapting the same ordering on the Rust side as here however, but I think you argued against that ordering in a previous comment?

Reply:
I meant just use it as an option for the Wasm bindings (and the canonical default is `FailFast::AllErrors` in my opinion).

I argued that the `options` should be at the end because creating them inline obscures any parameters that follow it.

I think `(presentation, options, fail_fast, holder, issuers)` would be acceptable for now, the main point is that Rust and Wasm should match.

Reply:
I will adapt Rust to also take `(presentation, options, fail_fast, holder, issuers)`.

Reply:
Cool. Are we still dead-set on not having `FailFast` in `PresentationValidationOptions` and `CredentialValidationOptions`? I don't think it's too difficult to handle and the behaviour would make sense:
- If unset default to `FailFast::AllErrors`.
- If set on presentation options and not on credential options use for both presentation and credential validations.
- If set on credential options and not on presentation options use `FailFast::AllErrors` on presentation and the set value for the credential validations.
- If set on both, use their respective values in presentation validation and credential validation separately.

Not a big deal either way.

Reply:
I get that semanitcally it makes sense to bundle `FailFast` with the validation options, but it is both simpler and more robust to keep it a separate parameter. In this case I would prefer to favour the latter.

Reply:
I would leave it up to @olivereanderson what to do here, but my preference would also be with @cycraig's suggestion. Mainly because we could get rid of the additional parameter, and the signature is fairly verbose with 5 parameters. Just a slight preference.

Reply:
Five parameters is not that much when keeping in mind all the work this method does. Part of the reason for changing the type of `fail_fast` from `bool` to an enum was to make it more transparent to the caller and I feel placing it in the validation options takes some of that benefit away. Since none of you expressed a very strong preference here I think I will keep this the way it is for now.

Reply:
The same can be said for any of the validation options, however. Developers are unlikely to conditionally change `FailFast` from the default in general as it's really only an optimisation.